### PR TITLE
Support matching media type for posts

### PR DESCRIPTION
**Pull request by charleso**
@reiddraper As hinted at, something seems missing in the current design. This was just one idea, I'm not at all sure it's correct. I do wonder whether changing `processPost :: Webmachine m [(MediaType, PostResponse m)]` would be better. How would that affects forwarding to a `PUT`.

Thoughts? Suggestions?

### CODE REVIEW

**Comment by charleso** [comment]
Actually I didn't know (or forgot) about `knownContentType`. More unsure/confused. :(

**Comment by tmcgilchrist** [comment]
The closet thing I could see in the webmachine repo is this open issue.
https://github.com/webmachine/webmachine/issues/122

Obviously the types for PostResponse mean our issue looks slightly different but it still comes down to when handling a POST airship provide a way to do content-type negotiation on the POST body. The workaround is treating the POST as a PUT and first generating an ID for it `PostCreate "some-url"`and then process the post inside `contentTypesAccepted`.

**Comment by reiddraper** [comment]
Given that we've implemented this now for PATCH, I think we should definitely go through with implementing it for POST. @charleso you mind giving it a rebase?

**Comment by charleso** [comment]
@reiddraper Happy to, although probably won't get a chance until this weekend. Do the media types look right on `PostProcess`/`PostProcessRedirect`, and not the other two?

File: src/Airship/Resource.hs [hunk 1]
```diff
@@ -23,11 +23,16 @@ import           Network.HTTP.Types
 -- | Used when processing POST requests so as to handle the outcome of the binary decisions between
 -- handling a POST as a create request and whether to redirect after the POST is done.
 -- Credit for this idea goes to Richard Wallace (purefn) on Webcrank.
+--
+-- For processing the POST, an association list of 'MediaType's and 'Webmachine' actions are required
+-- that correspond to the accepted @Content-Type@ values that this resource can accept in a request body.
+-- If a @Content-Type@ header is present but not accounted for, processing will halt with
+-- @415 Unsupported Media Type@.
```
Comment (charleso) [inline src/Airship/Resource.hs:30]: Have added some doc here, I can't English so someone might want to check this

File: src/Airship/Internal/Decision.hs [hunk 2]
```diff
@@ -95,12 +95,9 @@ newtype IfNoneMatch = IfNoneMatch ByteString
 -- Decision Helpers
 ------------------------------------------------------------------------------
 
-negotiateContentTypesAccepted :: Monad m => Resource m -> FlowStateT m ()
-negotiateContentTypesAccepted Resource{..} = do
+negotiateContentTypesAccepted :: Monad m => [(MediaType, Webmachine m a)] -> FlowStateT m a
+negotiateContentTypesAccepted accepted = do
```
Comment (charleso) [inline src/Airship/Internal/Decision.hs:99]: Have tweaked this to make it more re-usable. I'm about 99.99% sure there isn't a code path that is now passing in `contentTypesAccepted` that was previously using `patchContentTypesAccepted`. Will need to be careful when `o14` or similar is wired in.

**Comment by charleso** [comment]
Have updated. Let me know if it looks OK.

**Pull request merged**